include/arpa/nameser.h

```diff
@@ -308,15 +309,15 @@ typedef enum __ns_cert_types {
 #define NS_PUT16(s, cp) ns_put16((s), ((cp)+=2)-2)
 #define NS_PUT32(l, cp) ns_put32((l), ((cp)+=4)-4)
 
-unsigned ns_get16(const unsigned char *);
-unsigned long ns_get32(const unsigned char *);
-void ns_put16(unsigned, unsigned char *);
-void ns_put32(unsigned long, unsigned char *);
+unsigned ns_get16(const unsigned char * : count(2));
+unsigned long ns_get32(const unsigned char * : count(4));
+void ns_put16(unsigned, unsigned char * : count(2));
+void ns_put32(unsigned long, unsigned char * : count(4));
 
-int ns_initparse(const unsigned char *, int, ns_msg *);
-int ns_parserr(ns_msg *, ns_sect, int, ns_rr *);
-int ns_skiprr(const unsigned char *, const unsigned char *, ns_sect, int);
-int ns_name_uncompress(const unsigned char *, const unsigned char *, const unsigned char *, char *, size_t);
+int ns_initparse(const unsigned char * : count(len), int len, ns_msg * : itype(ptr<ns_msg>));
+int ns_parserr(ns_msg * : itype(ptr<ns_msg>), ns_sect, int, ns_rr * : itype(ptr<ns_rr>));
+int ns_skiprr(const unsigned char * cp : bounds(cp, eom), const unsigned char * eom : itype(ptr<const unsigned char>), ns_sect, int);
```

Comment:
Adding bounds safe interface to function signatures seems to affect readability. Could you please have each parameter on a new line to help readability?